tcs/csp_solver_mspt_receiver.cpp

```diff
@@ -619,422 +650,169 @@ void C_mspt_receiver::call(const C_csp_weatherreader::S_outputs &weather,
-						* I_bn*field_eff_adj*m_A_sf / 1000. / (CSP::pi*m_h_rec*m_d_rec / (double)n_flux_x);	//[kW/m^2];
-				}
-			}
-		}
-		else
-		{
-			m_flux_in.fill(0.0);
-		}
+	soln.dni = I_bn;
+	soln.field_eff = field_eff;
+	soln.T_salt_cold_in = T_salt_cold_in;	// Cold salt inlet temperature (K)
+	soln.od_control = m_od_control;         // Initial defocus control (will be adjusted during the solution)
+	soln.mode = m_mode;
+	soln.itermode = m_itermode;
+	soln.rec_is_off = rec_is_off;
 
-		double n_flux_x_d = (double)m_n_flux_x;
-		double n_panels_d = (double)m_n_panels;
+	clearsky = get_clearsky(weather, hour);
+	double clearsky_adj = std::fmax(clearsky, weather.m_beam);   // Set clear-sky DNI to actual DNI if actual value is higher
 
-		if( m_n_panels >= m_n_flux_x )
-		{
-			// Translate to the number of panels, so each panel has its own linearly interpolated flux value
-			for( int i = 0; i < m_n_panels; i++ )
-			{
-				double x = (i + 0.5)*n_flux_x_d / n_panels_d;
-				double xlow = ceil(i*n_flux_x_d / n_panels_d) - 0.5;
-				int high = ceil(i*n_flux_x_d / n_panels_d);
-				int low = high - 1;
-				if (high > m_n_flux_x - 1) high = 0;
-				if (low < 0) low = m_n_flux_x - 1;
-				double psp_field = m_flux_in.at(low) + (m_flux_in.at(high) - m_flux_in.at(low)) * (x - xlow);  //[kW/m^2] Average area-specific power for each node
-
-				/*
-				double ppos = (n_flux_x_d / n_panels_d*i + n_flux_x_d*0.5 / n_panels_d);
-				int flo = (int)floor(ppos);
-				int ceiling = (int)ceil(ppos);
-				double ind = (int)((ppos - flo) / fmax((double)(ceiling - flo), 1.e-6));
-				if( ceiling > m_n_flux_x - 1 ) ceiling = 0;
-
-				double psp_field = (ind*(m_flux_in.at(ceiling) - m_flux_in.at(flo)) + m_flux_in.at(flo));		//[kW/m^2] Average area-specific power for each node
-				*/
-				m_q_dot_inc.at(i) = m_A_node*psp_field;	//[kW] The power incident on each node
 
-			}
-		}
-		else
-		{
-			/*
-			The number of panels is always even, therefore the receiver panels are symmetric about the N-S plane.
+	if (rec_is_off)
+		soln.q_dot_inc.resize_fill(m_n_panels, 0.0);
 
-			The number of flux points may be even or odd. The distribution is assumed to be symmetric
-			about North, therefore:
-			(a) A distribution with an odd number of points includes a center point (n_flux_x - 1)/2+1
-			whose normal faces exactly north
-			(b) A distribution with an even number of points includes 2 points n_flux_x/2, n_flux_x/2+1
-			which straddle the North vector.
-			In either scenario, two points straddle the South vector and no scenario allows a point to fall
-			directly on the South vector. Hence, the first and last flux points fall completely on the first
-			and last panel, respectively.
-			*/
-
-			double leftovers = 0.;
-			int index_start = 0; int index_stop = 0;
-			double q_flux_sum = 0.0;
+	else
+	{
+		//--- Solve for mass flow at actual and/or clear-sky DNI extremes
+		if (m_csky_frac <= 0.9999 || fabs(I_bn - clearsky_adj) < 0.001)  // Solve for mass flow at actual DNI?
```

Comment:
Similar for m_csky_frac -- should probably be clearsky_frac_   (Again, pedantic, and don't need to change. Just looking to persuade changing some of our coding practices.) We're currently using "m_" prefix for the member variables, so following local convention is good, but I'd really like to move away from this antiquated convention.

Reply:
(note the underscore after clearsky_frac_ )